src/widgets/modals/PostDetailModal.tsx

```diff
@@ -0,0 +1,53 @@
+import { CommentType } from "@entities/comment/model/types"
+import { Post } from "@entities/post/model/types"
+import { highlightText } from "@shared/lib/highlightText"
+import { Dialog, DialogContent, DialogHeader, DialogTitle } from "@shared/ui"
+import { CommentSection } from "@widgets/comments"
+
+interface PostDetailModalProps {
+  comments: CommentType[]
+  isOpen: boolean
+  onAddComment: (postId: number) => void
+  onClose: () => void
+  onDeleteComment: (id: number, postId: number) => void
+  onEditComment: (comment: CommentType) => void
+  onLikeComment: (id: number, postId: number) => void
+  searchQuery: string
+  selectedPost: Post | null
+}
+
+export const PostDetailModal = ({
+  comments,
+  isOpen,
+  onClose,
+  selectedPost,
+  searchQuery,
+  onAddComment,
+  onDeleteComment,
+  onEditComment,
+  onLikeComment,
+}: PostDetailModalProps) => {
+  if (!isOpen || !selectedPost) return null
+
+  return (
+    <Dialog onOpenChange={onClose} open={isOpen}>
+      <DialogContent className="max-w-3xl">
+        <DialogHeader>
+          <DialogTitle>{highlightText(selectedPost.title, searchQuery)}</DialogTitle>
```

Comment:
소소하지만 컴파운드 패턴은 어떠신지도 제안드려봅니다.

```suggestion
    <Dialog onOpenChange={onClose} open={isOpen}>
      <Dialog.Content className="max-w-3xl">
        <Dialog.Header>
          <Dialog.Title>{highlightText(selectedPost.title, searchQuery)}</DialogTitle>
```